client/src/state/products/productInfoSlice.js

```diff
@@ -0,0 +1,37 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios'; 
+
+const initialState = {
+  data: {}, 
+  isLoading: false, 
+  error: null,
+};
+
+export const fetchProductInfo = createAsyncThunk(
+  'productInfo/fetchProductInfo', 
+  async (productId) => {
+    const res = await axios.get(`/products/${productId}`);
+    return res.data; 
+  }
+)
+
+const productInfoSlice = createSlice({
+  name: 'productInfo', 
+  initialState, 
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchProductInfo.pending, (state) => {
+      state.isLoading = true
+    });
+    builder.addCase(fetchProductInfo.fulfilled, (state, action) => {
+      state.isLoading = false
+      state.data = {...state.data, [action.payload.id]: action.payload }
+    });
+    builder.addCase(fetchProductInfo.rejected, (state, action) => {
+      state.isLoading = false
+      state.error = action.error.message
+    });
```

Comment:
this file i would need an explanation but it looks like this is just the way to add the data from the axios call to the store?

Reply:
Yes, when fetchProductInfo is dispatched redux toolkit uses the extra reducers defined here to handle different states of the dispatched event created by createAsyncThunk. So when it is fulfilled the defined callback on line 26 is called and redux's state updated with the data returned from the function defined in fetchProductInfo (attached to the action.payload).